core/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -144,10 +150,11 @@
         .put(OBJECT.id(), ImmutableSet.of(GEO_SHAPE))
         .put(ArrayType.ID, ImmutableSet.of()) // convertability handled in ArrayType
         .put(SetType.ID, ImmutableSet.of()) // convertability handled in SetType
+        .put(SingleColumnTableType.ID, ImmutableSet.of()) // convertability handled in SetType
```

Comment:
I guess it is possible that a single column table might be cast to an array. This line doesn't add any support for casting since it returns an empty list.